src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -173,17 +183,23 @@ private void removeDeadDataNodes() {
     return allTasksIds.values();
   }
 
-  public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType) {
+  public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType, String taskName) {
     switch (taskType) {
       case HDFSConstants.NAME_NODE_ID :
         HashMap<String, String> nameNodes = getNameNodes();
         nameNodes.put(hostname, taskId.getValue());
         setNameNodes(nameNodes);
+        HashMap<String, String> nameNodeTaskNames = getNameNodeTaskNames();
+        nameNodeTaskNames.put(taskId.getValue(), taskName);
+        setNameNodeTaskNames(nameNodeTaskNames);
         break;
       case HDFSConstants.JOURNAL_NODE_ID :
         HashMap<String, String> journalNodes = getJournalNodes();
         journalNodes.put(hostname, taskId.getValue());
         setJournalNodes(journalNodes);
+        HashMap<String, String> journalNodeTaskNames = getJournalNodeTaskNames();
+        journalNodeTaskNames.put(taskId.getValue(), taskName);
```

Comment:
lets use the task name as the key instead. That way we don't have to do `.values()` and we reinforce the idea of unique task name